src/NuGet.Server.Core/Core/DuplicatePackageException.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGet.Server.Core
+{
+    /// <summary>
+    /// This exception is thrown when trying to add a package in a version that already exists on the server 
+    /// and <see cref="NuGet.Server.Core.Infrastructure.ServerPackageRepository.AllowOverrideExistingPackageOnPush"/> is set to false.
+    /// </summary>
+    public class DuplicatePackageException : InvalidOperationException
```

Comment:
Does this need to extend `InvalidOperationException`? I wouldn't extend `InvalidOperationException` unless necessary

Reply:
I did this thinking that it still is an invalid operation thus I wanted to "keep it in there". But I will remove it if you wish.